pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -119,3 +132,76 @@ def visit_for(self, node):
                     continue
                 self.add_message("consider-using-enumerate", node=node)
                 return
+
+    def _check_consider_using_dict_items(self, node: astroid.For) -> None:
+        """Add message when accessing dict values by index lookup."""
+        # Verify that we have a .keys() call and
+        # that the object which is iterated is used as a subscript in the
+        # body of the for.
+
+        iterating_object_name = utils.get_iterating_dictionary_name(node)
+        if iterating_object_name is None:
+            return
+
+        # Verify that the body of the for loop uses a subscript
+        # with the object that was iterated. This uses some heuristics
+        # in order to make sure that the same object is used in the
+        # for body.
+        for child in node.body:
+            for subscript in child.nodes_of_class(astroid.Subscript):
+                subscript = cast(astroid.Subscript, subscript)
+
+                if not isinstance(subscript.value, (astroid.Name, astroid.Attribute)):
```

Comment:
```suggestion
                subscript = cast(astroid.Subscript, subscript)
                if not isinstance(subscript.value, (astroid.Name, astroid.Attribute)):
```
If you what to, this would be possible

Reply:
Changed in 797b050